crates/turborepo-lockfiles/src/pnpm/data.rs

```diff
@@ -270,6 +272,19 @@ impl PnpmLockfile {
         name: &str,
         specifier: &'a str,
     ) -> Result<Option<&'a str>, crate::Error> {
+        // Handle catalog references (https://pnpm.io/catalogs)
+        if specifier.starts_with("catalog:") {
+            if let Some(catalogs) = &self.catalogs {
+                let catalog_name = specifier.strip_prefix("catalog:").unwrap_or("default");
+                if let Some(catalog) = catalogs.get(catalog_name) {
+                    if let Some(dep) = catalog.get(name) {
+                        return Ok(Some(&dep.version));
+                    }
+                }
+            }
+            return Ok(None);
```

Comment:
Just using `strip_prefix` from the jump lets us skip the `unwrap`
```suggestion
        if let Some(catalog_name) = specifier.strip_prefix("catalog:") {
            if let Some(catalogs) = &self.catalogs {
                if let Some(catalog) = catalogs.get(catalog_name) {
                    if let Some(dep) = catalog.get(name) {
                        return Ok(Some(&dep.version));
                    }
                }
            }
            return Ok(None);
```